platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/NativeMapView.java

```diff
@@ -381,14 +381,14 @@ public void setZoom(double zoom) {
     if (isDestroyedOn("setZoom")) {
       return;
     }
-    setZoom(zoom, 0);
+    setZoom(zoom, -1, -1, 0);
   }
 
-  public void setZoom(double zoom, long duration) {
+  public void setZoom(double zoom, double cx, double cy, long duration) {
     if (isDestroyedOn("setZoom")) {
       return;
     }
-    nativeSetZoom(zoom, duration);
+    nativeSetZoom(zoom, cx == -1 ? cx : cx / pixelRatio, cy == -1 ? cy : cy / pixelRatio, duration);
```

Comment:
What about extracting `cx == -1 ? cx : cx / pixelRatio, cy == -1 ? cy : cy / pixelRatio` into a query or a explaining variable?
It's really hard to read it and therefore to understand what's going on.